lib/Sema/TypeCheckDecl.cpp

```diff
@@ -868,6 +868,12 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
   if (!currentFile || currentDC->isLocalContext())
     return;
 
+  // Cannot define a type called 'Any'
+  if (current->getFullName() == tc.Context.Id_Any) {
```

Comment:
Ideally `Any` is a TypeExpr and `` `Any` `` is a DeclRefExpr that is treated like any other identifier (i.e. it does not necessarily resolve to the magic Any).